### PR TITLE
Amqp dropped ack fix

### PR DESCRIPTION
**Pull request by timtay-microsoft**

### CODE REVIEW

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothubservices/TransportClientTests.java [hunk 1]
```diff
@@ -340,7 +340,7 @@ public void sendMessagesOverAmqpsWsMultithreaded() throws InterruptedException,
                             deviceListAmqps[i], clientArrayList.get(i), NUM_MESSAGES_PER_CONNECTION, NUM_KEYS_PER_MESSAGE, SEND_TIMEOUT_MILLISECONDS, cdl))
                     .start();
         }
-        cdl.await();
+        cdl.await(3, TimeUnit.MINUTES);
```
Comment (timtay-microsoft) [inline iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothubservices/TransportClientTests.java]: >await [](start = 12, length = 5)

These tests aren't failing at this point, but if they ever did, they would run for our entire 15 minute test timeout which is unacceptable.

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsDeviceAuthenticationCBS.java [hunk 2]
```diff
@@ -115,7 +115,17 @@ void sendAuthenticationMessages() throws TransportException
                     }
                 }
                 // Codes_SRS_AMQPSDEVICEAUTHENTICATIONCBS_12_009: [The function shall set the delivery tag for the sender.]
-                byte[] deliveryTag = String.valueOf(this.nextTag++).getBytes();
+                byte[] deliveryTag = String.valueOf(this.nextTag).getBytes();
```
Comment (jasmineymlo) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsDeviceAuthenticationCBS.java]: nit: Refactor and reuse the same logic is used in AmqpsSessionDeviceOperation.java

**Comment by jasmineymlo** [comment]
reminder to run e2e against Basic Tier

**Pull request merged**